ddtrace/contrib/trace_utils.py

```diff
@@ -595,3 +597,54 @@ def set_user(tracer, user_id, name=None, email=None, scope=None, role=None, sess
             "See https://docs.datadoghq.com/security_platform/application_security/setup_and_configure/"
             "?tab=set_user&code-lang=python for more information.",
         )
+
+
+def _start_span_internal(
+    tracer,  # type: Tracer
+    name,  # type: str
+    integration_config,  # type: IntegrationConfig
+    child_of=None,  # type: Optional[Union[Span, Context]]
+    service=None,  # type: Optional[str]
+    resource=None,  # type: Optional[str]
+    span_type=None,  # type: Optional[str]
+    activate=False,  # type: bool
+    meta=None,  # type: Optional[Dict[str, str]]
+    metrics=None,  # type: Optional[Dict[str, float]]
+):
+    span = tracer.start_span(name, child_of, service, resource, span_type, activate)
+    span.set_tag_str("component", integration_config.integration_name)
+
+    if meta:
+        for k, v in iter(meta.items()):
+            span.set_tag_str(k, v)
+
+    if metrics:
+        for k, v in iteritems(metrics):
+            tracer.set_metric(k, v)
+
+    return span
+
+
+def _trace_internal(
+    tracer,  # type: Tracer
+    name,  # type: str
+    integration_config,  # type: IntegrationConfig
+    service=None,  # type: Optional[str]
```

Comment:
We can get the service name from the integration config. I don't think this argument is required